### DISCUSSION TITLE
Writing custom Schema.NET classes

### DISCUSSION THREAD

**Discussion by kirk-marple** [RehanSaeed/Schema.NET#147] [Q&A]
### Question

To date, I've been using the standard Schema.NET classes, and I came across pending support for PodcastEpisode on schema.org.  In my project, I'm supporting podcast shows and episodes, and currently using RadioSeries/Episode to store the data.

But I was wondering if it's common for folks using Schema.NET to build your own classes, derived from existing Schema.NET classes, to support pending schema.org data models, or even just custom JSON-LD for your own use?

I have another case of handling YouTube channels and playlists, and currently using MusicPlaylist, but would rather have a custom VideoPlaylist class.

Not a huge deal, but when serializing to JSON-LD, it would be a lot cleaner to have it say 'VideoPlaylist' not 'MusicPlaylist', for example.

### Schema Objects

Which [schema.org](https://schema.org) object is this about if any?

https://schema.org/PodcastEpisode

**Comment by RehanSaeed** [comment]
Not sure if it's common but some developers have reported that they are writing custom types inherited from the standard schema.org types.

This scenario is supported and in fact we've added features in more recent versions to make it simpler to do. If you find any issues, we'd be glad to fix them.

I hope that answers your question!
